### PR TITLE
add convention for plugin location (relates to #13879)

### PR DESCRIPTION
**Pull request by AmiStrn**
### Description
adding convention for where plugin code should be

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/13879

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CONVENTIONS.md
Comment (andrross) [inline CONVENTIONS.md:20]:
```suggestion
- Most plugins should start in a repository outside of the OpenSearch project. This allows for rapid iteration during initial development without the overhead of being an official part of the project.
```

File: CONVENTIONS.md
Comment (andrross) [inline CONVENTIONS.md:22]:
```suggestion
- Do not add new plugins to the [plugins folder](https://github.com/opensearch-project/OpenSearch/tree/main/plugins) within the core OpenSearch repository. The maintainers likely do not have domain expertise in your new optional plugin. If you think it should be there anyway, please explain why in the RFC stage. In any case, it is recommended to start outside the project.   
```

File: CONVENTIONS.md
Comment (andrross) [inline CONVENTIONS.md:18]: Need to add this to the table of contents

**Review by andrross — COMMENTED** [review]
A few wording nitpicks, but overall I like it

**Comment by AmiStrn** [comment]
> A few wording nitpicks, but overall I like it

Thanks @andrross :) 
accepted the changes

**Review by reta — APPROVED** [review]

**Comment by AmiStrn** [comment]
awaiting a second reviewer: @dblock @VachaShah @andrross @saratvemulapalli

**Review by andrross — APPROVED** [review]

**Comment by AmiStrn** [comment]
oh sorry @andrross i didn't see you are not on the maintainer list for this repo. thanks for the review though. 
@dblock
@VachaShah
@saratvemulapalli
@ohltyler

**Review by saratvemulapalli — APPROVED** [review]

**Comment by andrross** [comment]
> oh sorry @andrross i didn't see you are not on the maintainer list for this repo

And here I thought you just really valued my opinion :)

**Comment by AmiStrn** [comment]
> > oh sorry @andrross i didn't see you are not on the maintainer list for this repo
> 
> And here I thought you just really valued my opinion :)

lol, that is NOT how I wished to come across. sarcasm aside (some have told me I am incapable of sarcasm), your opinion is very much valued!

File: CONVENTIONS.md
Comment (dblock) [inline CONVENTIONS.md:21]: Nit: Remove most. It's good to be directive. 
Nit: "OpenSearch project" -> opensearch-project ...
Nit: "start" is probably not right, "create" is more appropriate.

I would say "Create new plugins in your personal GitHub account. ..."

File: CONVENTIONS.md
Comment (dblock) [inline CONVENTIONS.md:22]: Ported -> moved
OpenSearch Project -> opensearch-project

File: CONVENTIONS.md
Comment (dblock) [inline CONVENTIONS.md:23]: "In any case, it is recommended to start outside the project. " is kinda redundant.

File: CONVENTIONS.md
Comment (dblock) [inline CONVENTIONS.md:19]: Should this be "Creating New Plugins"?

**Review by dblock — APPROVED** [review]
I made some nitpicky comments, feel free to use/ignore/resolve.

File: CONVENTIONS.md
Comment (AmiStrn) [reply CONVENTIONS.md:19]: Modifying this gives the title room to expand for future issues if that is what you mean.

**Review by dblock — APPROVED** [review]

**Comment by AmiStrn** [comment]
need one other approval for this one 🙏

**Comment by dblock** [comment]
I pinged @VachaShah for #229.

**Pull request merged**
